examples/chip-tool/commands/common/DeviceScanner.h

```diff
@@ -0,0 +1,70 @@
+/*
+ *   Copyright (c) 2023 Project CHIP Authors
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ */
+
+#pragma once
+
+#include <controller/CHIPDeviceController.h>
+#include <lib/dnssd/ResolverProxy.h>
+
+#include <vector>
+
+#if CONFIG_NETWORK_LAYER_BLE
+#if CHIP_DEVICE_LAYER_TARGET_DARWIN
+#include <platform/Darwin/BleScannerDelegate.h>
+#define USE_BLE_SCANNER 1
```

Comment:
Maybe call it CHIP_TOOL_DEVICE_SCANNER_USE_BLE just to avoid accidental collisions with something else?